File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.java
class Solution {
    public int maxOperations(int[] nums, int k) {
        
        HashMap <Integer,Integer> box=new HashMap<>();

        int res=0;

        for(int i=0;i<nums.length;i++){
            if(box.containsKey(nums[i]) && box.get(nums[i])>=1){
                res++;
                box.put(nums[i],box.getOrDefault(nums[i],0)-1);
            }
            else if(!box.containsKey(k-nums[i])){
                box.put(k-nums[i],1);
            }
            else if(box.containsKey(k-nums[i])){
               box.put(k-nums[i],box.getOrDefault(k-nums[i],0)+1);
            }
        }
        return res;
    }
}

File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:

        nums=sorted(nums)
        l=0
        r=len(nums)-1
        res=0
        while l<r:
            if nums[l]+nums[r]==k:
                print(nums[l],nums[r])
                res+=1
            l+=1
            r-=1
        return res
        